3.analysis/scripts/3.t_tests.py

```diff
@@ -216,5 +202,236 @@
 f_statistic, p_value = f_oneway(adult_male_latent_df_float, ped_male_latent_df_float, adult_female_latent_df_float, ped_female_latent_df_float)
 anova_df = pd.DataFrame({'f_stat': f_statistic.tolist(), 'p_value': p_value.tolist()})
 anova_df['latent_feature'] = anova_df.index + 1
+# Remove rows with NaN values
+anova_df = anova_df.dropna()
 anova_df
 
+
+# In[60]:
+
+
+#Lung Cancer in adult vs Neuroblastoma in ped comparison
+NB_ids = model_df.query("OncotreePrimaryDisease == 'Neuroblastoma'").ModelID.tolist()
+
+ped_NB_latent_df = ped_latent_df.copy()
+for index, row in ped_NB_latent_df.iterrows():
+    if row['ModelID'] not in NB_ids:
+        ped_NB_latent_df.drop(index, inplace=True)
```

Comment:
It looks like you're trying to filter rows in your DataFrame that match the NB_ids in the ModelID column, and discard the rest. If that's correct, and assuming NB_ids is a list, you can simplify and speed up your code by using the .isin() method.

This approach not only reduces the lines of code but also significantly improves performance. Here's an example:

(Feel free to disregard if I've misunderstood your intent!)
```suggestion
ped_NB_latent_df = ped_latent_df[ped_latent_df['ModelID'].isin(NB_ids)].copy()
```
This approach is much faster and more efficient, especially for large datasets.

Reply:
thanks!